bootstrap.sh

```diff
@@ -1,48 +1,383 @@
+if [ "$(id -u)" = 0 ]; then
+	edebug "Script has been executed as user with ID 0, assuming root"
+	funcname="$myName"
+	return 0
+# NOTICE(Krey): The ID 33333 is used by gitpod
+elif [ "$(id -u)" = 1000 ] || [ "$(id -u)" = 33333 ]; then
+	ewarn "Script $myName is not expected to run as non-root, trying to elevate root.."
+	if command -v sudo 1>/dev/null; then
+		einfo "Found 'sudo' that can be used for root elevation"
+		rootEle="sudo"
+		funcname="$myName"
+		return 0
+	elif command -v su 1>/dev/null; then
+		einfo "Found 'su' that can be used for a root elevation"
+		ewarn "This will require the end-user to parse a root password multiple times assuming that root has a password set"
+		rootEle="su -c"
+		funcname="$myName"
+		return 0
+	elif ! command -v sudo 1>/dev/null && ! command -v su 1>/dev/null; then
+		die 3 "Script $myName depends on root permission to install packages where commands 'sudo' nor 'su' are available for root elevation"
+		funcname="$myName"
+		return 0
+	else
+		die 225 "processing root on non-root"
+	fi
+else
+	die 3 "Unknown user ID '$(id -u)' has been parsed in script $myName"
 fi
-set -x
 
-sudo ninja install -C builddir
+if command -v "$UNAME" 1>/dev/null; then
+	unameKernel="$("$UNAME" -s)"
+	edebug "Identified the kernel as '$unameKernel"
+	case "$unameKernel" in
+		Linux)
+			KERNEL="$unameKernel"
 
-# Restart polkit so we don't get pop-ups whenever we pkexec
-if systemctl list-unit-files |grep -q polkit.service; then
-    sudo systemctl try-restart polkit
+			# Assume Linux Distro and release
+			# NOTICE(Krey): We are expecting this to return a lowercase value
+			if command -v "$LSB_RELEASE" 1>/dev/null; then
+				assumedDistro="$("$LSB_RELEASE" -si | "$TR" :[upper]: :[lower]:)"
+				assumedRelease="$("$LSB_RELEASE" -cs | "$TR" :[upper]: :[lower]:)"
+			elif ! command -v "$LSB_RELEASE" 1>/dev/null && [ -f /etc/os-release ]; then
+				assumedDistro="$("$GREP" -o "^ID\=.*" /etc/os-release | "$SED" s/ID=//gm)"
+				assumedRelease="$("$GREP" -o"^VERSION_CODENAME\=.*" /etc/os-release | "$SED" s/VERSION_CODENAME=//gm)"
+			elif ! command -v "$LSB_RELEASE" 1>/dev/null && [ ! -f /etc/os-release ]; then
+				die 1 "Unable to identify linux distribution using  command 'lsb_release' nor file '/etc/os-release'"
+			else
+				die 255 "attempting to assume linux distro and release"
+			fi
+
+			edebug "Identified distribution as '$assumedDistro'"
+			edebug "Identified distribution release as '$assumedRelease'"
+
+			# Verify Linux Distro
+			efixme "Add sanitization logic for other linux distributions"
+			case "$assumedDistro" in
+				ubuntu | debian | fedora | nixos | opensuse | gentoo | exherbo)
+					DISTRO="$assumedDistro"
+				;;
+				*) die fixme "Unexpected Linux distribution '$assumedDistro' has been detected."
+			esac
+
+			# Verify Linux Distro Release
+			efixme "Sanitize verification of linux distro release"
+			assumedRelease="$RELEASE"
+		;;
+		FreeBSD | Redox | Darwin | Windows)
+			KERNEL="$unameKernel"
```

Comment:
Useless, gamemode won't run on any of those since they don't have systemd or even elogind and probably will never have.